spec/controllers/chargeback_controller_spec.rb

```diff
@@ -113,27 +113,28 @@
   end
 
   context "#get_cis_all" do
-    elem_objects = [:storage, :miq_enterprise, :ext_management_system, :ems_cluster, :tenant]
-    elem_objects.each do |obj|
-      let!(obj) { FactoryGirl.create(obj) }
+    let!(:storage) { FactoryGirl.create(:storage) }
+    let!(:miq_enterprise) { FactoryGirl.create(:miq_enterprise) }
+
+    it "returns names of instances of enterprise" do
+      names_miqent = {}
+      MiqEnterprise.all.each do |instance|
+        names_miqent[instance.id] = instance.name
+      end
+      controller.instance_variable_set(:@edit, :new => {:cbshow_typ => "enterprise"}, :cb_assign => {})
+      controller.send(:get_cis_all)
+      expect(assigns(:edit)[:cb_assign][:cis]).to eq(names_miqent)
     end
-    elements_miq = %w(enterprise storage ext_management_system ems_cluster tenant)
-    elements_miq.each do |element|
-      classtype =
-        if element == "enterprise"
-          MiqEnterprise
-        else
-          element.classify.constantize
-        end
-      it "returns names of instances of " + element do
-        names = {}
-        classtype.all.each do |instance|
-          names[instance.id] = instance.name
-        end
-        controller.instance_variable_set(:@edit, :new => {:cbshow_typ => element}, :cb_assign => {})
-        controller.send(:get_cis_all)
-        expect(assigns(:edit)[:cb_assign][:cis]).to eq(names)
+
+    it "returns names of instances of storage" do
+      names_storage = {}
+      element = "storage"
+      element.classify.constantize.all.each do |instance|
```

Comment:
@aljesusg there can be directly Storage.all... instead of converting it from string